src/mixins/java/org/spongepowered/common/mixin/core/world/entity/item/PrimedTntMixin.java

```diff
@@ -129,7 +136,8 @@ public abstract class PrimedTntMixin extends EntityMixin implements PrimedTntBri
 
     @Inject(method = "tick()V", at = @At("RETURN"))
     private void impl$updateTNTPushPrime(final CallbackInfo ci) {
-        if (this.shadow$getFuse() == this.bridge$fuseDuration - 1 && !this.shadow$level().isClientSide) {
+        if (!impl$postPrimeTriggered && !this.shadow$level().isClientSide) {
```

Comment:
What was this change for? Also remember the this qualifier (and other places).

Reply:
I made this change because every time you change the fuse duration, the post prime event would get called

Reply:
Ah, right. You also changed that the current fuse time is changed when then duration is modified.

I think the general idea here is that you can set the fuse duration independently of the current remaining ticks. So lets say we have already ticked 5 times and now you change the fuse duration to 10. That would now mean we explode after 5 more ticks instead of 10.

Reply:
Wouldn't you also hit this when you set the remaining ticks to the fuse duration? So keeping the `impl$postPrimeTriggered` around would still be beneficial?

Reply:
Oh yeah, that's true! Sorry!